src/Invoice.php

```diff
@@ -219,4 +222,58 @@ public function setFooterText($footerText)
         $this->footerText = $footerText;
         return $this;
     }
+
+    public function getFinancialCode()
+    {
+        return $this->financialCode;
+    }
+
+    public function setFinancialCode($financialCode)
+    {
+        $this->financialCode = $financialCode;
+	    return $this;
+    }
+
+    public function getFinancialNumber()
+    {
+        return $this->financialNumber;
+    }
+
+    public function setFinancialNumber($financialNumber)
+    {
+        $this->financialNumber = $financialNumber;
+	    return $this;
+    }
+
+    /**
+     * References the office in which the transaction was booked.
+     */
+    public function getOffice(): Office
+    {
+        return $this->office;
+    }
+
+    /**
+     * References the daybook on which the transaction was booked.
+     */
+    public function getCode(): string
+    {
+        return $this->financialCode;
+    }
+
+    /**
+     * References the transaction.
+     */
+    public function getNumber(): int
+    {
+        return $this->financialNumber;
+    }
+
+    /**
+     * Reference the exact line in the transaction.
+     */
+    public function getLineId(): int
+    {
+        return 1;
```

Comment:
Unfortunately, the 2nd argument in the `MatchLine::addToMatchSet`-method is typehinted to the `MatchReferenceInterface` interface, which `BookingReferenceInterface` isn't and does not inherit in some way.

Reply:
@Thijs-Riezebeek can we change / tweak this?

Reply:
If this works for you @Wouter0100 then I guess it's okay. Looking at the Twinfield API docs the transaction line number does not seem optional.

Reply:
@Wouter0100 @willemstuursma You cannot match with a `BookingReferenceInterface`, since you have to match specific lines in the Twinfield API. The `BookingReferenceInterface` is only used as a reference to an entire booking (e.g. a BankTransaction, instead of a specific line in that BankTransaction).

Reply:
When matching invoices, you always want to match the TOTAL line of the invoice. Twinfield wants the TOTAL line to always be at the top of the `<lines>`, so the TOTAL line should always have line ID 1, so I think that this will work, although it does feel a bit hacky.

I'm not too familiar with the `Invoice` class in this package, though, as we've only used the `SalesTransaction` for booking invoices.

Reply:
That is exactly what I discovered. I think it wasn't documentation, but I think the Twinfield support confirmed that this would be the right solution.